pkg/resources/endpoints.go

```diff
@@ -53,3 +35,74 @@ func ParentResourceFromService(name string) string {
 	}
 	return name[:li]
 }
+
+// ReadyPodCounter provides a count of currently ready pods. This
+// information is used by UniScaler implementations to make scaling
+// decisions. The interface prevents the UniScaler from needing to
+// know how counts are performed.
+// The int return value represents the number of pods that are ready
+// to handle incoming requests.
+// The error value is returned if the ReadyPodCounter is unable to
+// calculate a value.
+type ReadyPodCounter interface {
+	ReadyCount() (int, error)
+}
+
+type scopedEndpointCounter struct {
+	endpointsLister corev1listers.EndpointsLister
+	namespace       string
+	serviceName     string
+}
+
+func (eac *scopedEndpointCounter) ReadyCount() (int, error) {
+	return fetchReadyAddressCount(eac.endpointsLister, eac.namespace, eac.serviceName)
+}
+
+// NewScopedEndpointsCounter creates a ReadyPodCounter that uses
+// a count of endpoints for a namespace/serviceName as the value
+// of ready pods. The values returned by ReadyCount() will vary
+// over time.
+// lister is used to retrieve endpoints for counting with the
+// scope of namespace/serviceName.
+func NewScopedEndpointsCounter(lister corev1listers.EndpointsLister, namespace, serviceName string) ReadyPodCounter {
+	return &scopedEndpointCounter{
+		endpointsLister: lister,
+		namespace:       namespace,
+		serviceName:     serviceName,
+	}
+}
+
+type unscopedCounter struct {
+	endpoints *corev1.Endpoints
+}
+
+func (uc *unscopedCounter) ReadyCount() (int, error) {
+	return readyAddressCount(uc.endpoints), nil
+}
+
+// NewFixedEndpointsListCounter creates a ReadyPodCounter that counts
+// the provided list of Endpoints. The value returned by ReadyCount()
+// should be stable, unless the endpoints parameter is modified elsewhere.
+func NewFixedEndpointsListCounter(endpoints *corev1.Endpoints) ReadyPodCounter {
```

Comment:
```suggestion
func NewFixedEndpointsCounter(endpoints *corev1.Endpoints) ReadyPodCounter {
```

For consistency with the above.